doc/source/reference/arrays.promotion.rst

```diff
@@ -0,0 +1,263 @@
+
+However, these rules can also lead to surprising behaviors when working with
+low precision dtypes.
+
+First, since the Python value is converted to a NumPy one before the operation
+can by performed, operations can fail with an error when the result seems
+obvious. For instance, ``np.int8(1) + 1000`` cannot continue because ``1000``
+exceeds the maximum value of an ``int8``. When the Python scalar
+cannot be coerced to the NumPy dtype, an error is raised:
+
+  >>> np.int8(1) + 1000
+  Traceback (most recent call last):
+    ...
+  OverflowError: Python integer 1000 out of bounds for int8
+  >>> np.int64(1) * 10**100
+  Traceback (most recent call last):
+  ...
+  OverflowError: Python int too large to convert to C long
+  >>> np.float32(1) + 1e300
+  np.float32(inf)
+  ... RuntimeWarning: overflow encountered in cast
+
+Second, since the Python float or integer precision is always ignored, a low
+precision NumPy scalar will keep using its lower precision unless explicitly
+converted to a higher precision NumPy dtype or Python scalar (e.g. via ``int()``,
+``float()``, or ``scalar.item()``). This lower precision may be detrimental to
+some calculations or lead to incorrect results, especially in the case of integer
+overflows:
+
+  >>> np.int8(100) + 100  # the result exceeds the capacity of int8
+  np.int8(-56)
+  ... RuntimeWarning: overflow encountered in scalar add
+
+Note that NumPy warns when overflows occur for scalars, but not for arrays;
+e.g., ``np.array(100, dtype="uint8") + 100`` will *not* warn.
+
+Numerical promotion
+-------------------
+
+The following images shows the numerical promotion rules with the kinds
+on the vertical axis and the precision on the horizontal axis.
+
+.. figure:: figures/nep-0050-promotion-no-fonts.svg
+    :figclass: align-center
+
+Promotion is always found along the lines in the schema from left to right
+and down to up.  The final result dtype will smallest kind and precision
+(i.e. down and left) which is still equal or higher in both kind and precision
+while following the (green) lines.
+
+Note the following specific rules and observations:
+1. When a Python ``float`` or ``complex`` interacts with a NumPy integer
+   the result will be ``float64`` or ``complex128`` (yellow border).
+   NumPy booleans will also be cast to the default integer.[#default-int]
+   This is not relevant when additionally NumPy floating point values are
+   involved.
+2. The precision is drawn such that ``float16 < int16 < uint16`` because
+   large ``uint16`` do not fit ``int16`` and large ``int16`` will lose precision
+   when stored in a ``float16``.
+   This pattern however is broken since NumPy always considers ``float64``
+   and ``complex128`` to be acceptable promotion results for any integer
+   value.
+3. A special case the above leads to is that NumPy will promote many
+   combinations of signed and unsigned integers to ``float64`` because no
+   integer dtype can hold both inputs.
+   This can unfortunately be 
```

Comment:
Incomplete sentence

Reply:
Just removed it, I think this is already a lot for a niche detail. (slight tweak above)